connectors/src/connectors/confluence/index.ts

```diff
@@ -0,0 +1,218 @@
+          dataSourceName: dataSourceConfig.dataSourceName,
+        },
+        { transaction }
+      );
+      await ConfluenceConfigurations.create(
+        {
+          cloudId,
+          connectorId: connector.id,
+          url: cloudUrl,
+        },
+        { transaction }
+      );
+
+      return connector;
+    });
+
+    // TODO(2024-01-10 flav) Uncomment in next PR.
+    // await launchConfluenceFullSyncWorkflow(connector.id, null);
+
+    return new Ok(connector.id.toString());
+  } catch (e) {
+    logger.error({ error: e }, "Error creating confluence connector.");
+    return new Err(e as Error);
+  }
+}
+
+export async function updateConfluenceConnector(
+  connectorId: ModelId,
+  {
+    connectionId,
+  }: {
+    connectionId?: NangoConnectionId | null;
+  }
+): Promise<Result<string, ConnectorsAPIErrorResponse>> {
+  console.log({ connectorId, connectionId });
+  throw new Error("Not implemented");
+}
+
+export async function retrieveConfluenceConnectorPermissions({
+  connectorId,
+  parentInternalId,
+  filterPermission,
+}: Parameters<ConnectorPermissionRetriever>[0]): Promise<
+  Result<ConnectorResource[], Error>
+> {
+  if (parentInternalId) {
+    return new Err(
+      new Error(
+        "Confluence connector does not support permission retrieval with `parentInternalId`"
+      )
+    );
+  }
+
+  const connector = await Connector.findOne({
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!connector) {
+    logger.error({ connectorId }, "Connector not found");
+    return new Err(new Error("Connector not found"));
+  }
+
+  const confluenceConfig = await ConfluenceConfigurations.findOne({
+    where: {
+      connectorId: connectorId,
+    },
+  });
+  if (!confluenceConfig) {
+    logger.error({ connectorId }, "Confluence configuration not found");
+    return new Err(new Error("Confluence configuration not found"));
+  }
+
+  const confluenceConnection = await getConnectionFromNango({
+    connectionId: connector.connectionId,
+    integrationId: getRequiredNangoConfluenceConnectorId(),
+    useCache: false,
+  });
+
+  const { access_token: confluenceAccessToken } =
+    confluenceConnection.credentials;
+
+  const spaces = await listConfluenceSpaces(
+    confluenceAccessToken,
+    confluenceConfig.cloudId
+  );
+
+  const syncedSpaces = await ConfluenceSpaces.findAll({
+    where: {
+      connectorId: connectorId,
+    },
+  });
+
+  const allSpaces: ConnectorResource[] = spaces.map((s) => {
+    const isSynced = syncedSpaces.some((ss) => ss.spaceId === s.id);
+
+    return {
+      provider: "confluence",
+      internalId: s.id,
+      parentInternalId: null,
+      type: "folder",
+      title: `${s.name}`,
+      sourceUrl: `${confluenceConfig.url}/wiki${s._links.webui}`,
+      expandable: false,
+      permission: isSynced ? "read" : "none",
+      dustDocumentId: null,
+      lastUpdatedAt: null,
+    };
+  });
+
+  // List synced spaces.
+  if (filterPermission === "read") {
```

Comment:
Can we not do a call to Confluence if we're in filterPermission = read to speed up a notch the selection of resources in assistants?

Reply:
Fixed.